src/Polyfills/AssertStringContains.php

```diff
@@ -63,6 +78,54 @@ public static function assertStringNotContainsString( $needle, $haystack, $messa
 	 * @return void
 	 */
 	public static function assertStringNotContainsStringIgnoringCase( $needle, $haystack, $message = '' ) {
+		// Replicate fix added to PHPUnit 6.4.2.
+		if ( self::needsEmptyStringFix( $needle, true ) ) {
+			self::markTestSkipped( 'Asserting a string contains an empty string, which always "exists" in any other string.');
+		}
+
 		static::assertNotContains( $needle, $haystack, $message, true );
 	}
+
+	/**
+	 * Decide if a fix for an empty needle string is needed.
+	 *
+	 * PHPUnit 6.4.2 added a fix to gracefully handle the case when an empty string was passed to
+	 * `assertStringContainsString()` and variant assertions. Versions earlier than that gave a "mb_strpos(): Empty
+	 * delimiter" error instead.
+	 *
+	 * If the needle string is empty, and the PHPUnit version is found to be lower than 6.4.2, or 6.4.2 or above and
+	 * part of a NotContains assertion, then return true, which will result in tests marked as skipped.
+	 *
+	 * To not have tests marked as skipped, test authors should ensure the needle string
+	 * passed to the assertion is not empty.
+	 *
+	 * @param string $needle_string String that should be looked for in haystack string.
+	 * @param bool   $is_negative   Whether assertion is negative i.e. NotContains
+	 *
+	 * @return bool True unless the needle string is empty, or PHPUnit is 6.4.2 or later, or
+	 *              PHPUnit version can't be identified.
+	 */
+	protected static function needsEmptyStringFix( $needle_string, $is_negative ) {
```

Comment:
If this method remains, it can be simplified a bit.

Two options:
1. If you'd only be looking for "is the version above 6.x.x.", you can use something along the lines of the below:
    ```php
    if ( \class_exists( '\PHPUnit\Runner\Version' ) === true && \version_compare( Version::id(), '6.4.2', '>=' ) ) {}
    ```
    ... as if the `\PHPUnit\Runner\Version` class doesn't exist, we already know it will be PHPUnit < 6.0.0.
    This is similar to patterns as used in various places in the `phpunitpolyfills-autoload.php` file.
2. Alternatively, you could add a class alias for the `PHPUnit_Runner_Version` class to the new name in the `Autoload::loadAssertStringContains()` method.
    This is already done for various other PHPUnit classes in the `phpunitpolyfills-autoload.php` file for those polyfills which needed it.

Reply:
Oh and another suggestion - if the `markTestSkipped()` remains, the code in the assertions can be simplified by moving the call to `markTestSkipped()` to this method and just doing a plain, non-conditional call to it from the assertion methods.
The method name should in that case probably be changed to something like `skipOnEmptyNeedle()`.

Reply:
Edit: Updated code snippet (due to some upcoming changes):

```php
if ( \class_exists( '\PHPUnit_Runner_Version' ) === true || \version_compare( Version::id(), '6.4.2', '>=' ) ) {}
```